src/VoodooClient.ts

```diff
@@ -0,0 +1,117 @@
+import xmtpv3 from 'xmtpv3'
+
+// TODO: this is a hacky wrapper class for a Voodoo contact,
+// currently represented by the entire contact's VoodooInstance
+// - should be changed to align with VoodooPublicAccount (or whatever)
+// it ends up being named in Rust
+export class VoodooContact {
+  address: string
+  voodooInstance: any
+
+  constructor(address: string, voodooInstance: any) {
+    this.address = address
+    this.voodooInstance = voodooInstance
+  }
+}
+
+/**
+ * Class that acts as a Client stand-in, but with Voodoo logic
+ * - sessions
+ * - contacts
+ * - encoding and decoding messages
+ *
+ * NOTE: there is a lot of duplication with Client.ts, but by
+ * making a dedicated class we can keep the logic clean for
+ * our own reasoning. Client.ts will consume the VoodooManager
+ * for now.
+ */
+export default class VoodooClient {
+  address: string
+  voodooInstance: any
+  // For now contacts are a map of address to VoodooInstance, all local
+  contacts: Map<string, VoodooContact> = new Map()
+
+  constructor(address: string, voodooInstance: any) {
+    this.address = address
+    this.voodooInstance = voodooInstance
+  }
+
+  static async create(address: string): Promise<VoodooClient> {
+    const xmtpWasm = await xmtpv3.XMTPWasm.initialize()
+    // TODO: STARTINGTASK: this just creates unused voodoo keys that go nowhere
+    const myVoodoo = xmtpWasm.newVoodooInstance()
+    return new VoodooClient(address, myVoodoo)
+  }
+
+  setContact(address: string, contactInfo: VoodooContact) {
+    this.contacts.set(address, contactInfo)
+  }
+
+  getContact(address: string): VoodooContact | undefined {
+    return this.contacts.get(address)
+  }
+
+  // Get the JSON from creating an outbound session with a contact
+  async getOutboundSessionJson(
+    contactAddress: string,
+    initialMessage: string
+  ): Promise<string> {
+    // Get the contact info which is just a handle for now
+    const contactInstance = this.contacts.get(contactAddress)?.voodooInstance
+    if (!contactInstance) {
+      throw new Error(`No contact info for ${contactAddress}`)
+    }
+    const outbound = await this.voodooInstance.createOutboundSession(
+      contactInstance,
+      initialMessage
+    )
+    return outbound
```

Comment:
Won't we need to also get the `session_id` to be able to do anything with this?

Reply:
Yup, we need to add some session tracking when we integrate `encrypt` and `decrypt`. Otherwise all we can do is what I did in my crappy prototype: decrypt the very first message